### PR TITLE
pyuvdata integration_time and polarization convention update

### PR DESCRIPTION
**Pull request by jsdillon**
I've closed #359 because it was failing tests and my pushes to that branch to fix them were not showing up on the PR.

This PR addresses #348.

### CODE REVIEW

File: hera_cal/io.py [hunk 1]
```diff
@@ -741,6 +739,10 @@ def write_vis(fname, data, lst_array, freq_array, antpos, time_array=None, flags
     time_array = np.repeat(time_array[np.newaxis], Nbls, axis=0).ravel()
     lst_array = np.repeat(lst_array[np.newaxis], Nbls, axis=0).ravel()
 
+    # configure integration time
+    if integration_time is None:
+        integration_time = np.ones_like(time_array, dtype=np.float) * np.median(np.diff(time_array)) * 24 * 3600.
```
[the remark this answers is not in the document]
Comment (jsdillon) [reply hera_cal/io.py]: Done.

File: hera_cal/lstbin.py [hunk 2]
```diff
@@ -599,6 +599,11 @@ def lst_bin_files(data_files, dlst=None, verbose=True, ntimes_per_file=60, file_
     time bins. Output miriad file meta data (frequency bins, antennas positions, time_array)
     are taken from the first file in data_files.
 
+    Note: Only supports input data files that have nsample_array == 1, and a single
```
[the remark this answers is not in the document]
Comment (jsdillon) [reply hera_cal/lstbin.py]: Added an assertion.

File: hera_cal/lstbin.py [hunk 3]
```diff
@@ -673,7 +678,7 @@ def lst_bin_files(data_files, dlst=None, verbose=True, ntimes_per_file=60, file_
     antpos = copy.deepcopy(ap)
     start_jd = np.floor(t)[0]
     kwargs['start_jd'] = start_jd
-    kwargs['integration_time'] = np.median(np.diff(t)) * 24 * 3600.
+    integration_time = np.median(np.diff(t)) * 24 * 3600.
```
[the remark this answers is not in the document]
Comment (jsdillon) [reply hera_cal/lstbin.py]: Done.

File: hera_cal/lstbin.py [hunk 4]
```diff
@@ -673,12 +673,13 @@ def lst_bin_files(data_files, dlst=None, verbose=True, ntimes_per_file=60, file_
...
     d, fl, ap, a, f, t, l, p = io.load_vis(data_files[0][0], return_meta=True, pick_data_ants=False)
 
-    # get frequency, time and antenna position information
+    # get frequency, time (converting times in days to integration times in seconds) and antenna position information
     freq_array = copy.copy(f)
     antpos = copy.deepcopy(ap)
     start_jd = np.floor(t)[0]
     kwargs['start_jd'] = start_jd
     integration_time = np.median(np.diff(t)) * 24 * 3600.
+    assert np.all(np.diff(t) == np.median(np.diff(t))), 'All integrations must be of equal length (BDA not supported).'
```
Comment (plaplant) [inline hera_cal/lstbin.py]: I'd rather have this be a proper `if` clause that raises an error if it's not true (I'm not a huge fan of plain `assert` statements in general...), but we can leave it if you prefer.

**Comment by jsdillon** [comment]
And thank you for the quick turnaround @plaplant.

**Pull request merged**